src/hotspot/share/opto/chaitin.cpp

```diff
@@ -1727,6 +1727,13 @@ void PhaseChaitin::fixup_spills() {
           if( cisc->oper_input_base() > 1 && mach->oper_input_base() <= 1 ) {
             assert( cisc->oper_input_base() == 2, "Only adding one edge");
             cisc->ins_req(1,src);         // Requires a memory edge
+          } else {
+            // There is no space reserved for a memory edge before the inputs for
+            // instructions which have "stackSlotX" parameter instead of "memory".
+            // For example, "MoveF2I_stack_reg". We always need a memory edge from
+            // src to cisc, else we might schedule cisc before src, loading from a
+            // spill location before storing the spill.
+            cisc->add_prec(src);
```

Comment:
An assertion for null memory operand can be added before adding presidence edge.

Reply:
@jatin-bhateja Thanks, I added it :)

Reply:
`memory_operand()` returns a `MachOper*`, we should not compare it to `0` but to `nullptr` (see https://github.com/openjdk/jdk/blob/master/doc/hotspot-style.md).

Reply:
@TobiHartmann thanks for the feedback, I had assumed it had the same return as `int InstructForm::memory_operand`, but that is not the case. Will change it.